### PR TITLE
Log 729 over the wire updates

### PR DESCRIPTION
**Pull request by DanielSlavov**

### CODE REVIEW

File: linux-package/application.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+organizationId: ba2cbc90-5424-11e8-b88d-6f2c1b6625e8
```
Comment (Glamdring) [inline linux-package/application.yaml:1]: this file should not be needed

Comment (Glamdring) [reply linux-package/application.yaml:1]: remove it, and use a maven ant script to copy the one from src/main/resources, just as in the Windows case

File: linux-package/setup-collector-centos.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# install Java 9
+sudo yum -y update
+sudo yum -y remove java
+
+rpm -ivh /tmp/install/jdk-9_linux-x64_bin.rpm
```
Comment (Glamdring) [inline linux-package/setup-collector-centos.sh:7]: instead use 
yum -y install java-11-openjdk-devel
and do not remove existing java packages

File: linux-package/update-collector.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# cd /var/logsentinel
+cd /home/daniel/Desktop
```
Comment (Glamdring) [inline linux-package/update-collector.sh:4]: /var/logsentinel-collector

File: src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    @PostConstruct
+    public void update() {
+        File lockFile = null;
+        String fileLocation = null;
+        try {
+            if (isUpdateAvailable()) {
+                currentlyUpdating = true;
+                String url = "https://github.com/LogSentinel/logsentinel-collector/releases/download/"
+                        + getLatestReleaseVersion() + "/logsentinel-agent.jar";
```
Comment (Glamdring) [inline src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java:45]: jar should be called -collector

File: src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+                if (System.getProperty("os.name").equals("Linux")) {
```
Comment (Glamdring) [inline src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java:47]: Prefer if (SystemUtils.IS_OS_LINUX)

File: src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            + "/logsentinel-collector/releases/latest";
+
+    private static final Logger logger = LoggerFactory.getLogger(CollectorUpdateService.class);
+
+    @Value("${build.version}")
+    private String currentVersion;
+
+    private boolean currentlyUpdating;
+
+    private static final String LINUX_LOCATION = "/home/daniel/Desktop/";
```
Comment (Glamdring) [inline src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java:35]: /var/logsentinel-collector

File: src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        ObjectNode rootNode = (ObjectNode) new ObjectMapper().readTree(response.getBody());
+
+        return rootNode.get("tag_name").textValue();
+    }
+
+    public boolean isUpdateAvailable() {
+        try {
+            return (currentVersion.compareTo(getLatestReleaseVersion()) < 0) && !currentlyUpdating;
+        } catch (Exception e) {
+            return false;
```
Comment (Glamdring) [inline src/main/java/com/logsentinel/agent/service/CollectorUpdateService.java:88]: log a warning

File: windows-installer/install.bat [hunk 8]
```diff
@@ -0,0 +1,31 @@
+@echo off
```
Comment (Glamdring) [inline windows-installer/install.bat:1]: I can't see from the pull request, but this is just a rename of windows-service -> windows-installer? You have to fix the pom.xml as well to account for that change (and reuse a similar plugin execution configuration to copy the linux files to target/linux-installer )

Comment (DanielSlavov) [reply windows-installer/install.bat:1]: I have also added update-collector.bat

**Pull request closed without merging**